### PR TITLE
Stellarators redefined simulate method

### PR DESCRIPTION
**Pull request by joaobernardo23**
This PR aims to include new arguments to the `simulate` method that will decide (when evaluated the API side) what is the simulation that will be performed. 

For the moment, I only included two cases, one where the user provides all the additional arguments and another where the user provides none of them. 

This seemed better for now rather than to check if every argument is None or not and then apply a default value to those that were, but it is easily extendible to that case, it would just make the code look a bit different with a lot of 'if' statements.

### CODE REVIEW

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 1]
```diff
@@ -230,15 +230,49 @@ def simulate(
...
         """Simulates the scenario.
 
+        A simulation can be performed in two ways simply by adding some more
+        arguments. 
+        
+        If `num_iterations`, `num_samples` and `sigma_scaling_factor`
+        are all not provided, the simulation merely uses the created device and 
+        performs the computation of the magnetic field produced by its coils
+        on the given plasma surface. Additionally, it produces an output of
+        the objective functions that assess the quality of the device.
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:246]: A simpler, more natural way to provide this functionality is by setting defaults to these parameters, namely:
```python
num_samples = 1
num_iterations = 1
sigma_scaling_factor = 0.1
```

Then explain what happens when `num_samples` and `num_iterations` > 1. WDYT?

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 2]
```diff
@@ -230,15 +230,49 @@ def simulate(
...
+
+        Alternatively, if the arguments are provided, the simulation process
+        is different. It also uses the create device, but as a beginning step.
+        Using this initial device, the simulation adds noise to the coefficients
+        that describe the coils using a normal distribution function, generating
+        `num_samples` configurations. After this, the configuration with the
+        lowest value of the objective functions is used as an initial step for
+        the next iteration, being that this process is repeated `num_iterations`
+        times. All the generated configurations (their Fourier Series
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:255]: "... used as an initial step for the next iteration. This process is repeated \`num_iterations\` times."

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 3]
```diff
@@ -250,14 +284,32 @@ def simulate(
...
+
+        if len(deciding_args) not in (0, 3):
+            raise ValueError('If you want to perform the stellarator noise '
+                             'search simulation please provide all 3 '
+                             'necessary arguments. These include:\n '
+                             '"num_iterations", "num_samples" and '
+                             '"sigma_scaling_factor".\nOtherwise, please do '
+                             'not provide any of these.')
+
+        else:
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:300]: See my comment above. I think we don't need this after all.

File: inductiva/stellarators/scenarios/coils/coils.py [hunk 4]
```diff
@@ -230,15 +230,49 @@ def simulate(
...
+        arguments. 
+        
+        If `num_iterations`, `num_samples` and `sigma_scaling_factor`
+        are all not provided, the simulation merely uses the created device and 
+        performs the computation of the magnetic field produced by its coils
+        on the given plasma surface. Additionally, it produces an output of
+        the objective functions that assess the quality of the device.
+
+        Alternatively, if the arguments are provided, the simulation process
+        is different. It also uses the create device, but as a beginning step.
```
Comment (fabiocruz) [inline inductiva/stellarators/scenarios/coils/coils.py:249]: We need to explain with 1/2 more sentences what this "search" process is about. Shall we use the word "optimization"?

**Review by fabiocruz — CHANGES_REQUESTED** [review]
Thanks @joaobernardo23. I think that setting the right default values could simplify a lot the logic of the scenario. See my comments below and let me know what you think.

File: inductiva/stellarators/scenarios/coils/coils.py (hunk 1 above)
Comment (joaobernardo23) [reply inductiva/stellarators/scenarios/coils/coils.py:246]: I totally agree, much simpler and much easier to understand, thank you!

**Review by fabiocruz — COMMENTED** [review]
Suggestion to improve the docstring below.

**Review by fabiocruz — APPROVED** [review]

**Pull request merged**